### PR TITLE
Document context

### LINKED ISSUE #580: Add docs for context

### PR DESCRIPTION
**Pull request by spicyj**
Fixes #580.

### CODE REVIEW

File: docs/docs/12-context.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+## Passing info automatically through a tree
+
+Suppose you have a structure like:
+
+```javascript
+var Button = React.createClass({
+  render: function() {
+    return (
+      <button style={{'{{'}}background: this.props.color}}>
```
Comment (krzysiek1507) [inline docs/docs/12-context.md]: Is this line ok?

File: docs/docs/12-context.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+In this example, we manually thread through a `color` prop in order to style the `Button` and `Message` components appropriately. Theming is a good example of when you might want an entire subtree to have access to some piece of information (a color). Using context, we can pass this through the tree automatically:
+
+```javascript{2-4,7,18,25-30,33}
+var Button = React.createClass({
+  contextTypes: {
+    color: React.PropTypes.string
+  },
+  render: function() {
+    return (
+      <button style={{'{{'}}background: this.context.color}}>
```
Comment (krzysiek1507) [inline docs/docs/12-context.md]: The same as above.

File: docs/docs/12-context.md (hunk 1 above)
Comment (spicyj) [inline docs/docs/12-context.md]: Yeah. Jekyll turns this into `{{`.

**Comment by brigand** [comment]
Maybe a mention about high order components to abstract context access? It sort of hints to it here

> try to isolate your use of context to a small area and avoid using the context API directly when possible

Or maybe in a separate tip to avoid bloating this doc.

**Pull request merged**